File: python/1-two-sums.cpp
# include <vector>
# include <unordered_map>
using namespace std;

class Solution {
public:
vector<int> twoSum(vector<int> nums, int target){
    unordered_map<int,int> temp;

    for(int i=0;i<nums.size();i++){
        int complement = target-nums[i];

        if (temp.find(complement) != temp.end()){
            return {temp[complement], i};
        }

        else{
            temp[nums[i]] = i;
        }
    }
    return {};
}
};

File: 1-twosums/1-two-sums.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        temp = {}

        for i,num in enumerate(num):
            complement = target-num

            if complement in temp:
                return [i, temp[complement]]

            else:
                temp[num]=i
        return False